src/ucp/api/ucp.h

```diff
@@ -513,6 +513,17 @@ enum ucp_dt_type {
 };
 
 
+/**
+ * @ingroup UCP_DATATYPE
+ * @brief UCP datatype attributes field mask.
+ *
+ * The enumeration allows querying a specific field in @ref ucp_datatype_attr_t.
+ */
+enum ucp_dt_attr_field {
+    UCP_DATATYPE_ATTR_FIELD_PACKED_SIZE = UCS_BIT(0) /**< packed datatype size */
```

Comment:
maybe rename it to `PACKED_CONTIG_SIZE`, as we can't use this API for anything other than contig types?

Reply:
Depends. If there is no plan to support this query for any other datatypes, probably we can rename it. If there is a longer term plans to support this type of query for other datatypes we shell use generic name.

Reply:
So it might be safer to keep it as is. `enum ucp_dt_type` lists _contig_, _strided_, _iov_ and _generic_. iov is not supported and this seems to make sense to me with regard to the UCP API. It looks like _strided_ is not really used, yet. _contig_ and _generic_ are supported.

Reply:
I'm a bit concerned that 0 is used as an "unsupported". Imho, using -1 makes more sense.

Reply:
I expected these concerns, but for simplicity's sake I chose `0` for unsupported, which fits well with `size_t`. And a datatype with size 0 does not make sense, right? If we want `-1` for unsupported, we should also add a `#define UCP_DATATYPE_UNSUPPORTED ((size_t)-1)` or `#define UCP_DATATYPE_SIZE_UNKNOWN ((size_t)-1)`. So I am also okay with changing unsupported from `0` to `-1`.

Reply:
Looking at MPI API as a reference https://www.open-mpi.org/doc/v4.0/man3/MPI_Type_size.3.php
If MPI cannot figure out what to do, it returns MPI_UNDEFINED in size which is mapped to https://github.com/open-mpi/ompi/blob/eb87378cd8c55c08fde666c03a04aa2811da395a/ompi/include/mpi.h.in#L545

I think MPI solution is not elegant either :)

Looking at our other "query" APIs, I think the function should return `ucs_status_t`. It is consistent approach across our query API and provides clean path to return errors.

Reply:
It is between `UCS_ERR_INVALID_PARAM` and `UCS_ERR_UNSUPPORTED`.

Reply:
I leave the choice to you. What about using `UCS_ERR_INVALID_PARAM` in case an IOV datatype (created by `ucp_dt_make_iov()`) is passed and `UCS_ERR_UNSUPPORTED` for the other case where the size of the generic datatype is not available yet? Or should we add something like `UCS_ERR_NOT_AVAILABLE` for the latter?

For my use case I would only check for `UCS_OK`.

Reply:
> I leave the choice to you. What about using `UCS_ERR_INVALID_PARAM` in case an IOV datatype (created by `ucp_dt_make_iov()`) is passed and `UCS_ERR_UNSUPPORTED` for the other case where the size of the generic datatype is not available yet? Or should we add something like `UCS_ERR_NOT_AVAILABLE` for the latter?
> 
> For my use case I would only check for `UCS_OK`.

I agree with above proposal.

Reply:
I added `ucs_status_t` as return type and changed the description accordingly.